### PR TITLE
[FEATURE]: QuizMaster Maintenance module frontend.

### PR DESCRIPTION
**Pull request by JMark-FS**
Pages
- questions (Question Table page)
- questions/create-question (Create Question Page)
- questions/edit/[questionId] (Edit Question Page of the given question Id)
- categories (Category Table Page)
- difficulties (Difficulty Table Page)
- dashboard (Dashboard page, BLANK)

Functionalites
- Search (Questions, Categories, Difficulties)
- Pagination (Question, Categories, Difficulties). 
- Add Question.
- Add Category.
- Add Difficulty.
- Remove Question.

Ongoing
- Edit Question
- Edit Category
- Edit Difficulty
- Remove Category
- Remove Difficulty

### CODE REVIEW

File: WebApp/backend/QuizMaster.API.Quiz/Controllers/QuestionCategoryController.cs [hunk 1]
```diff
@@ -25,11 +28,37 @@ public QuestionCategoryController(IQuizRepository quizRepository, IMapper mapper
...
+			if (resourceParameter.IsGetAll)
+			{
+				var categoriesFromDb = await _quizRepository.GetAllCategoriesAsync();
+
+				return Ok(_mapper.Map<IEnumerable<CategoryDto>>(categoriesFromDb));
+			}
 			// Get all active categories asynchronously
-			var categories = await _quizRepository.GetAllCategoriesAsync();
-			return Ok(_mapper.Map<IEnumerable<CategoryDto>>(categories));
+			var categories = await _quizRepository.GetAllCategoriesAsync(resourceParameter);
```
Comment (jaymar921) [inline WebApp/backend/QuizMaster.API.Quiz/Controllers/QuestionCategoryController.cs:40]: nice pagination feature

**Review by jaymar921 — APPROVED** [review]
Approved

**Review by jumapaoA — APPROVED** [review]
Approved

**Comment by jaymar921** [comment]
This PR can now be merged

**Pull request merged**